steveluscher/solana-playground
# feature: add an alias for the Technology Preview of the new web3.js

# Summary

By no means a complete solution, this PR is the start of adding support for the Technology Preview version of `@solana/web3.js`.

Here, we import the `tp` version of the library and alias it to `web3tp`.

# Test Plan

1. Create a wallet and airdrop some SOL to it.
2. Create a client like so:

   ```ts
   import "@solana/webcrypto-ed25519-polyfill";

   // Client
   const rpc = web3tp.createSolanaRpc({
     transport: web3tp.createDefaultRpcTransport({
       url: "https://api.devnet.solana.com",
     }),
   });

   const keyPair = await web3tp.generateKeyPair();
   console.log(keyPair);

   console.log("My address:", pg.wallet.publicKey.toString());
   const { value: balance } = await rpc
     .getBalance(pg.wallet.publicKey.toBase58())
     .send();
   console.log(`My balance: ${Number(balance) / web3.LAMPORTS_PER_SOL} SOL`);
   ```

3. Observe output
   ```
   Running client...
     client.ts:
       { privateKey:
      { algorithm: { name: 'Ed25519' },
        extractable: false,
        type: 'private',
        usages: [ 'sign' ] },
     publicKey:
      { algorithm: { name: 'Ed25519' },
        extractable: true,
        type: 'public',
        usages: [ 'verify' ] } }
       My address: 9kShTNxtemuHJzsXBmSPpS4o8vSdPAgBf9uEdAadm9j6
       My balance: 0.01 SOL
   ```
